### DISCUSSION TITLE
Quick find species in Create Record and Edit record.

### DISCUSSION THREAD

**Discussion by nishantnnb** [Mattk70/Chirpity-Electron#127] [Ideas]
Hello,
Thank you for this wonder software which is getting improved with each version released. 

Enhancement requests:
1) Currently to Create Record and Edit record, species list is provided in a dropdown list. This makes the search a little time consuming. 
Can you try to include quick search like ebird Checklist on mobile phone: 
https://support.ebird.org/en/support/solutions/articles/48000960508-ebird-mobile-tips-tricks#anchorQuickEntry (Quick find codes section).

2) A bit odd one but can be very useful:
Birdnet now allows to train negative samples. e.g. if bird is getting falsely detected as 'xyz', we can create a negative folder for xyz (-xyz) and use it in the training so that model will try to correct itself.
Hence, when if I edited a species in Chirpity which was detected by birdnet, somehow there should option to retain the detection by birdnet as - (negative) sample. And I should be able to export that with - in the file name. 

3) Export audio clips for all detections instead filtering each species independently. Along with same negative samples as per point 2.

4) I ran analysis on a folder. Then went to Records >>> View chart. From here, i can't easily go back to analysis screen. Explore Archives option is available. But it has additional filters on the top for apply date filter, location, etc hence this may not be the exact same screen from where i had started the analysis of the folder.   

5) Create record:
Easy entry for create record: I want to create records for 1 particular species with region length = 3 sec.  Option to set this and when I am exploring the spectrogram, focus window should always be of 3sec and when i press a keyboard key (which was set in settings - say "C" for create), software will keep on adding record for the set specie for the selected area of 3 sec on the spectrogram .  

sorry for too many asks.

Pls keep up the good work.
Thank you.

Regards,
Nishant

**Comment by Mattk70** [comment]
> Hello, Thank you for this wonder software which is getting improved with each version released.

Thanks, I am glad you think so!

> Enhancement requests:
> 
> 1. Currently to Create Record and Edit record, species list is provided in a dropdown list. This makes the search a little time consuming.
>    Can you try to include quick search like ebird Checklist on mobile phone:
>    https://support.ebird.org/en/support/solutions/articles/48000960508-ebird-mobile-tips-tricks#anchorQuickEntry (Quick find codes section).

When you open the Create/Edit record dialog box, the dropdown is in focus. You can type the name of the species you want, and the selected item will jump to that point in the list as you type.

> 2. A bit odd one but can be very useful:
>    Birdnet now allows to train negative samples. e.g. if bird is getting falsely detected as 'xyz', we can create a negative folder for xyz (-xyz) and use it in the training so that model will try to correct itself.
>    Hence, when if I edited a species in Chirpity which was detected by birdnet, somehow there should option to retain the detection by birdnet as - (negative) sample. And I should be able to export that with - in the file name.

I am not sure I follow. Since BirdNet only looks for a '-' in the folder (it doesn't care what the file name is), why not save the files you have corrected to a folder you create with a '-' at the beginning?

> 3. Export audio clips for all detections instead filtering each species independently. Along with same negative samples as per point 2.

> 4. I ran analysis on a folder. Then went to Records >>> View chart. From here, i can't easily go back to analysis screen. Explore Archives option is available. But it has additional filters on the top for apply date filter, location, etc hence this may not be the exact same screen from where i had started the analysis of the folder.

I'm going to look into this for a future version, but it's a low priority. It's worth noting that:
a) you won't find the records you are currently looking at in the archive or charts until you save them
b) you could launch a second instance of Chirpity to view the archive / charts, whilst running your analysis in the first window.

Principally because of b) I haven't prioritised this.

> 5. Create record:
>    Easy entry for create record: I want to create records for 1 particular species with region length = 3 sec.  Option to set this and when I am exploring the spectrogram, focus window should always be of 3sec and when i press a keyboard key (which was set in settings - say "C" for create), software will keep on adding record for the set specie for the selected area of 3 sec on the spectrogram .

So - I guess it's clear that you want to use Chirpity to assist your creation of a custom BirdNET classifier. I'll level with you. Since I have not heard even one account of anyone creating a superior classifier to the original BirdNET using the training tools offered, (negative classes included)  I am sceptical that these endeavours will succeed.

I believe you yourself have been attempting this for several months. How is it going - am I wrong based on your results to date?

It's my hunch that the sigmoid activation function used in the BirdNET model means it will *always* suggest species that sound similar to 'xyz'. Whilst I remain in my sceptical state of mind, I'm not inclined to invest time and effort into the development of custom classifiers. Sorry.
